syntax = "proto3";

package inventory;

service InventoryService {
  rpc CheckInventory(CheckInventoryRequest) returns (CheckInventoryResponse);
  rpc UpdateInventory(UpdateInventoryRequest) returns (UpdateInventoryResponse);
}

message CheckInventoryRequest {
  string product_id = 1;
}

message CheckInventoryResponse {
  int32 quantity = 1;
}

message UpdateInventoryRequest {
  string product_id = 1;
  int32 quantity = 2;
}

message UpdateInventoryResponse {
  bool success = 1;
}
